Source/UnitTests/Common/BitFieldViewTest.cpp

```diff
@@ -0,0 +1,658 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <gtest/gtest.h>
+
+#include "Common/BitFieldView.h"
+#include "Common/CommonTypes.h"
+#include "Common/EnumFormatter.h"
+
+enum class TestEnum : u64
+{
+  A,
+  B,
+  C,
+  D,
+};
+
+struct TestStruct
+{
+  u64 hex;
+
+  BFVIEW_M(hex, u64, 0, 64, full_u64);  // spans whole storage
+  BFVIEW_M(hex, s64, 0, 64, full_s64);  // spans whole storage
+
+  BFVIEW_M(hex, u64, 9, 3, regular_field_unsigned);   // a plain bitfield
+  BFVIEW_M(hex, u64, 9, 3, regular_field_unsigned2);  // Just the very same bitfield again
+  BFVIEW_M(hex, s64, 9, 3, regular_field_signed);     // Same bitfield, but different sign
+
+  BFVIEW_M(hex, s64, 30, 4, at_dword_boundary);  // goes over the boundary of two u32 values
+
+  BFVIEW_M(hex, s64, 15, 1, signed_1bit);  // allowed values: -1 and 0
+
+  BFVIEW_M(hex, bool, 63, 1, flag);
+
+  BFVIEW_M(hex, TestEnum, 16, 2, enum_1);
+  BFVIEW_M(hex, TestEnum, 48, 2, enum_2);
+
+  TestStruct() = default;
```

Comment:
Are these constructors needed? Why not use brace-initialization instead?

Reply:
Weird flex to code-review the unit test, lol.  Looks like the non-explicit constructor is sort of useful for the first TestStruct's usage in for loops, but every other one does not need it.